Tests/CreateSliceForDiagnostics/main.cpp

```diff
@@ -0,0 +1,371 @@
+    int is_per[AMREX_SPACEDIM];
+    for (int i = 0; i < AMREX_SPACEDIM; i++)
+        is_per[i] = 1;
+
+    // This defines a Geometry object for each level
+    Vector<Geometry> geom(nlevs);
+    geom[0].define(domain, &real_box, CoordSys::cartesian, is_per);
+    for (int lev = 1; lev < nlevs; lev++) {
+	geom[lev].define(amrex::refine(geom[lev-1].Domain(), ref_ratio[lev-1]),
+			 &real_box, CoordSys::cartesian, is_per);
+    }
+    Vector<BoxArray> ba(nlevs);
+    ba[0].define(domain);
+    
+    // break the BoxArrays at both levels into max_grid_size^3 boxes
+    for (int lev = 0; lev < nlevs; lev++) {
+        ba[lev].maxSize(max_grid_size);
+    }
+
+    Vector<DistributionMapping> dmap(nlevs);
+
+    // default cell-centered multifab //
+    Vector<std::unique_ptr<MultiFab> > mf(nlevs);
+    // node-based multifab //
+    Vector<std::unique_ptr<MultiFab> > rho_mf(nlevs);
+    // Ex - Edge based //
+    Vector<std::unique_ptr<MultiFab> > Ex_mf(nlevs);
+    // Ey - Edge based //
+    Vector<std::unique_ptr<MultiFab> > Ey_mf(nlevs);
+    // Ez - Edge based //
+    Vector<std::unique_ptr<MultiFab> > Ez_mf(nlevs);
+    // Bx - face centered //
+    Vector<std::unique_ptr<MultiFab> > Bx_mf(nlevs);
+    // By - face centered //
+    Vector<std::unique_ptr<MultiFab> > By_mf(nlevs);
+    // Bz - face centered //
+    Vector<std::unique_ptr<MultiFab> > Bz_mf(nlevs);
+
+    // note that current density is edge centered similar to E-Field and therefore for testing purposes it is not explicitly included here. 
+
+    for (int lev = 0; lev < nlevs; lev++) {
+        dmap[lev] = DistributionMapping{ba[lev]};
+        mf[lev].reset(new MultiFab(ba[lev], dmap[lev], ncomp, nghost));
+        rho_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{1,1,1}), dmap[lev], ncomp, nghost));
+        Ex_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{0,1,1}), dmap[lev], ncomp, nghost));
+        Ey_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{1,0,1}), dmap[lev], ncomp, nghost));
+        Ez_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{1,1,0}), dmap[lev], ncomp, nghost));
+        Bx_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{1,0,0}), dmap[lev], ncomp, nghost));
+        By_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{0,1,0}), dmap[lev], ncomp, nghost));
+        Bz_mf[lev].reset(new MultiFab(amrex::convert(ba[lev], IntVect{0,0,1}), dmap[lev], ncomp, nghost));
+    }
+
+    Vector<MultiFab*> mf_ptr = GetVecOfPtrs(mf);
+    Vector<MultiFab*> rho_mf_ptr = GetVecOfPtrs(rho_mf);
+    Vector<MultiFab*> Ex_mf_ptr = GetVecOfPtrs(Ex_mf);
+    Vector<MultiFab*> Ey_mf_ptr = GetVecOfPtrs(Ey_mf);
+    Vector<MultiFab*> Ez_mf_ptr = GetVecOfPtrs(Ez_mf);
+    Vector<MultiFab*> Bx_mf_ptr = GetVecOfPtrs(Bx_mf);
+    Vector<MultiFab*> By_mf_ptr = GetVecOfPtrs(By_mf);
+    Vector<MultiFab*> Bz_mf_ptr = GetVecOfPtrs(Bz_mf);
+
+    InitializeVariables( mf_ptr, geom );
```

Comment:
Minor style note: you can actually do `InitializeVariables(GetVecOfPtrs(mf), geom);` and likewise for the create slice function, and avoid having to create the `mf_ptr` variable beforehand.

Reply:
Thank you. I have made this change.